src/hotspot/share/opto/parse2.cpp

```diff
@@ -1584,15 +1581,18 @@ void Parse::adjust_map_after_if(BoolTest::mask btest, Node* c, float prob,
 
   bool is_fallthrough = (path == successor_for_bci(iter().next_bci()));
 
-  if (path_is_suitable_for_uncommon_trap(prob)) {
+  if (path_is_suitable_for_uncommon_trap(prob) && (!OptimizeUnstableIf || !path->is_merged())) {
+    sync_jvms();
+    SafePointNode* sfpt = clone_map();
```

Comment:
Seems `sync_jvms()` and `clone_map() are only needed for `OptimizeUnstableIf`.
You need comment explaining why you added `!path->is_merged()` for `OptimizeUnstableIf` case.